contracts/BaseStrategy.sol

```diff
@@ -258,42 +276,76 @@ abstract contract BaseStrategy {
      * would be negatively affected if `harvest()` is not called shortly, then this can return `true`
      * even if the keeper might be "at a loss" (keepers are always reimbursed by yEarn)
      *
+     * NOTE: `callCost` must be priced in terms of `want`
+     *
      * NOTE: this call and `tendTrigger` should never return `true` at the same time.
      */
-    function harvestTrigger(uint256 gasCost) public virtual view returns (bool);
+    function harvestTrigger(uint256 callCost) public view returns (bool) {
+        StrategyParams memory params = vault.strategies(address(this));
+
+        // Should not trigger if strategy is not activated
+        if (params.activation == 0) return false;
+
+        // Should trigger if hadn't been called in a while
+        if (block.number.sub(params.lastReport) >= maxReportDelay) return true;
+
+        // If some amount is owed, pay it back
+        // NOTE: Since debt is adjusted in step-wise fashion, it is appropiate to always trigger here,
+        //       because the resulting change should be large (might not always be the case)
+        uint256 outstanding = vault.debtOutstanding();
+        if (outstanding > 0) return true;
+
+        // Check for profits and losses
+        uint256 total = estimatedTotalAssets();
+        if (total < params.totalDebt) return true; // We have a loss to report!
```

Comment:
this triggers for me a lot because of rounding errors. there should be a threshold added

for instance. minting a ctoken gives can give you a slight on paper loss due to rounding errors on converting exchange rate of ctoken back to underlying asset

Reply:
Very true, we can use the same trick with the gain and the return factor

Reply:
Fixed in 4c9d27c